### PR TITLE
QE: Multiple test framework refactors

### PR DESCRIPTION
**Pull request by nodeg**
⚠️ ⚠️ ⚠️ 
Please don't squash commits. This is a special branch that combine multiple refactors, so we prefer to keep a commit per PR merged into this branch, to facilitate navigating over the reasons for each change.
⚠️ ⚠️ ⚠️ 

## Prerequisites before merging this

- [x] https://github.com/openSUSE/twopence/pull/81
- [x] https://github.com/uyuni-project/sumaform/pull/1248

## What does this PR change?

- https://github.com/uyuni-project/uyuni/pull/6559
- https://github.com/uyuni-project/uyuni/pull/5743
- https://github.com/uyuni-project/uyuni/pull/9160
- https://github.com/uyuni-project/uyuni/pull/9198
- https://github.com/uyuni-project/uyuni/pull/9205
- https://github.com/uyuni-project/uyuni/pull/9214
- https://github.com/uyuni-project/uyuni/pull/9224
- https://github.com/uyuni-project/uyuni/pull/9229
- https://github.com/uyuni-project/uyuni/pull/9231
- https://github.com/uyuni-project/uyuni/pull/9240
- https://github.com/uyuni-project/uyuni/pull/9241
- https://github.com/uyuni-project/uyuni/pull/9243

## TODOs

- [x] include fixes for the GitHub Actions PR tests for using Ruby 3
- [ ] ...

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes
- [x] **DONE**

## Test coverage
- Cucumber tests were edited
- [x] **DONE**

## Links

Issue(s): #
Ports(s): # 
- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: .github/workflows/rubocop.yml [hunk 1]
```diff
@@ -17,19 +17,11 @@ jobs:
...
-        gem install parallel -v 1.10
-        gem install parser -v 3.0.0.0
-        gem install rainbow -v 2.2.2
-        gem install regexp_parser -v 1.8
-        gem install rexml
-        gem install rubocop-ast -v 1.7.0
-        gem install ruby-progressbar -v 1.7
-        gem install unicode-display_width -v 1.4.0
-        gem install rubocop -v 1.17.0
+        gem install rubocop -v 1.65.1
```
Comment (vandabarata) [inline .github/workflows/rubocop.yml:24]: I'm curious, why are the other gems no longer necessary in this section?

**Review by vandabarata — DISMISSED** [review]
Thank you so much for all this work

**Comment by srbarrios** [comment]
⚠️ 
Please let's keep it as draft, until master is unlock.
As this PR will contain the combination of multiple PRs merged into qe-develop branch.

**Comment by nodeg** [comment]
GitHub PR acceptance tests:
```bash
Run ./testsuite/podman_runner/05_install_gems_in_controller.sh
+ sudo -i podman exec controller bash -c 'export GEM_PATH="/usr/lib64/ruby/gems/3.3.0"'
+ sudo -i podman exec controller bash -c 'cd /testsuite && bundle.ruby3.3 install --gemfile Gemfile --verbose'
bash: bundle.ruby3.3: command not found
```

For this to work, we need to update the container. This however is not possible in this PR and would also affect the all other PRs. So for now, we cannot rely on the PR tests here.
I did run them in my fork, where
- all changes are made in the master branch
- then I built all containers
- triggered a PR with minor changes to see if the tests work with the new code: https://github.com/nodeg/uyuni/pull/36

**Comment by nodeg** [comment]
Ok, so with the changes as of now, we have [one error](https://reports.cucumber.io/reports/c7b15a30-57a4-4fdc-9477-594494363a11):
image
image

I will force push again so that the controller container will get updated only under certain conditions and not for every push to this PR like it is now.

**Comment by nodeg** [comment]
Rebase done.

File: testsuite/features/support/network_utils.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2024 SUSE LLC.
+# Licensed under the terms of the MIT license.
+
+require 'net/scp'
+require 'net/ssh'
+require 'stringio'
+
+Net::SSH::Transport::Algorithms::ALGORITHMS.each_value { |algs| algs.reject! { |a| a.match(/^ecd(sa|h)-sha2/) } }
+Net::SSH::KnownHosts::SUPPORTED_TYPE.reject! { |t| t.match(/^ecd(sa|h)-sha2/) }
```
Comment (nodeg) [inline testsuite/features/support/network_utils.rb:9]: I think you can use `=~` instead of `match()` again since we reverted to Ruby 3.1 again.

**Comment by nodeg** [comment]
The [last run](https://ci.suse.de/job/uyuni-master-dev-acceptance-tests-podman/1497/#showFailuresLink) in the Uyuni Podman CI showed the following errors:
image

The error message in all 3 cases is the same.
```bash
    Scenario: Create custom pillar to turn off "disable_local_repos"

When I turn off disable_local_repos for all clients

Message:

    No such file or directory @ rb_file_s_stat - /tmp/top.sls20240826-8907-lpgccg (Errno::ENOENT)
./features/support/network_utils.rb:53:in `scp_command'
./features/support/remote_node.rb:82:in `scp'
./features/support/remote_node.rb:181:in `inject'
./features/support/file_management.rb:60:in `file_inject'
./features/support/file_management.rb:82:in `inject_salt_pillar_file'
./features/step_definitions/salt_steps.rb:566:in `/^I install a salt pillar top file for "([^"]*)" with target "([^"]*)" on the server$/'
./features/step_definitions/command_steps.rb:42:in `/^I turn off disable_local_repos for all clients/'
features/core/srv_disable_local_repos_off.feature:10:in `I turn off disable_local_repos for all clients'
```

File: testsuite/features/support/network_utils.rb (hunk 2 above)
Comment (srbarrios) [reply testsuite/features/support/network_utils.rb:9]: If it works, let's keep, otherwise it will spoy again once we upgrade Ruby.

Comment (nodeg) [reply testsuite/features/support/network_utils.rb:9]: Sure, fine with me.

**Pull request merged**